R/mock-session.R

```diff
@@ -75,41 +75,149 @@ mapNames <- function(func, vals) {
   vals
 }
 
+#' Returns a noop implementation of the public method `name` of ShinySession.
+#' @include shiny.R
+#' @noRd
+makeNoop <- function(name, msg = paste0(name, " is a noop.")) {
+  if (!(name %in% names(ShinySession$public_methods)))
+    stop(name, " is not public method of ShinySession.")
+  impl <- ShinySession$public_methods[[name]]
+  body(impl) <- rlang::expr({
+    # Force arguments
+    !!lapply(formalArgs(impl), rlang::sym)
+    private$noopWarn(!!name, !!msg)
+    NULL
+  })
+  impl
+}
+
+#' Accepts a series of symbols as arguments and generates corresponding noop
+#' implementations.
+#' @noRd
+makeWarnNoops <- function(...) {
+  methods <- as.character(rlang::ensyms(...))
+  sapply(methods, makeNoop, USE.NAMES = TRUE, simplify = FALSE)
```

Comment:
Please use `lapply` -- in general, avoid `sapply` because it's not clear from looking at the code what type of object it returns (and also, the type can change depending on the input).

Reply:
Since lapply doesn't have a USE.NAMES option I switched to lapply and an assignment to names(), please let me know if you were suggesting something different